include/asm/rpn.h

```diff
@@ -14,7 +14,7 @@
 
 #include "linkdefs.h"
 
-#define MAXRPNLEN 1048576
+#define MAXRPNLEN 1048576LU
```

Comment:
But it has to be printed, and if I print it as `%u`, then it won't work on DOS, and if I print it as `%lu`, then it won't work on Linux/Mac/Windows.

Reply:
Wouldn't protected-mode DOS (DJGPP) have the same 32-bit `unsigned int` as 32-bit Windows? Or are you looking to build for real-mode DOS, 640 KiB limit and all?

Reply:
@pinobatch It will most likely be 16-bit for compatibility; `int` isn't required to be pointer-sized. (For comparison, `int` is 32 bits on most 64-bit platforms.)

Reply:
@pinobatch DJGPP still has a 16-bit `int`. That's actually why I'm doing this; I tried to compile with a DJGPP cross-compiler (latest GCC version at the time) and it spewed out a lot of warnings due to `printf` formats having the wrong type. This will fix those.

Reply:
Under DJGPP, `int` is indeed 32-bit. I downloaded DJGPP from [Zip Picker](http://www.delorie.com/djgpp/zip-picker.cgi), installed it into DOSBox, and keyed in this program:
```c
#include <limits.h>
#include <stdio.h>

int main(void) {
  printf("INT_MAX is %d\nUINT_MAX is %u\nCHAR_BIT is %u\nsizeof(int) is %u\n",
         INT_MAX, UINT_MAX, (unsigned)CHAR_BIT, (unsigned)sizeof(int));
  return 0;
}
```
Compiled it:
```
C:\>gcc -Wall -Wextra -O limits.c -o limits.exe
C:\>limits > out.txt
```
The result:
```
INT_MAX is 2147483647
UINT_MAX is 4294967295
CHAR_BIT is 8
sizeof(int) is 4
```

Reply:
@pinobatch Interesting. I got some warnings, and it doesn't hurt to increase portability anyways.